dot/network/helpers_test.go

```diff
@@ -18,15 +19,21 @@ import (
 )
 
 type testStreamHandler struct {
+	sync.Mutex
+
 	messages map[peer.ID][]Message
 	decoder  messageDecoder
-	exit     bool
+
+	// exitChan is a notify chan that the readStream
+	// works and sucessfuly returns
```

Comment:
Do you mean that the `readStream` function uses and closes when it returns? I'm a bit confused here

Reply:
yeah i'd just say "readStream closes exitChan when it returns"

Reply:
I think I forgot some words 😆 🤣